dymos/utils/introspection.py

```diff
@@ -607,19 +683,21 @@ def filter_outputs(patterns, sys):
         A dictionary where the matching output names are the keys and the associated dict provides
         the 'units' and 'shapes' metadata.
     """
-    outputs = sys if isinstance(sys, dict) else get_promoted_vars(sys, iotypes='output',
-                                                                  metadata_keys=['shape', 'units'])
+    outputs = sys if isinstance(sys, dict) else get_promoted_vars(sys, iotypes='output', metadata_keys=['shape', 'units', 'tags'])
+    _patterns = [patterns] if isinstance(patterns, str) else patterns
 
-    if '*' in patterns:  # shortcut for '*'
-        filtered = outputs
-    else:
-        filtered = set()
-        for pattern in patterns:
-            filtered.update(fnmatch.filter(outputs, pattern))
+    output_names = list(outputs.keys())
+    filtered = []
```

Comment:
Better to just make `filtered` a set to begin with and call `update` on it with results from fnmatch instead of creating a list first and then converting to a set.